app/file_sender.py

```diff
@@ -1,36 +1,129 @@
+    'P_IC_H2': 'Initial contact questionnaire households - Wales',
+    'P_IC_H4': 'Initial contact questionnaire households - Northern Ireland'
+}
+
+
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-def process_complete_file(file: Path):
-    # TODO Encrypt, create manifest, send to sftp, delete partial file
-    logger.info(f'doing the file stuff on {file}')
+def process_complete_file(file: Path, pack_code):
+    message = file.read_text()
+    # First encrypt the file and write it to encrypted directory
+    logger.info('Encrypting file', file_name=file.name)
+    encrypted_message = pgp_encrypt_message(message)
+    filename = f'{pack_code}_{datetime.utcnow().strftime("%Y-%m-%dT%H-%M-%S")}'
+    print_file = Config.ENCRYPTED_FILES_DIRECTORY.joinpath(f'{filename}.csv')
+    logger.info('Writing encrypted file', file_name=print_file.name)
+    with open(print_file, 'w') as encrypted_file:
+        encrypted_file.write(encrypted_message)
+
+    # Create the manifest
+    manifest_file = Config.ENCRYPTED_FILES_DIRECTORY.joinpath(f'{filename}.manifest')
+    logger.info('Creating manifest file', manifest_file=manifest_file.name)
+    generate_manifest_file(manifest_file, print_file, pack_code)
+    file_paths = [print_file, manifest_file]
+
+    # Send files to sftp
+    logger.info('Sending files to SFTP', file_paths=file_paths)
+    copy_files_to_sftp(file_paths)
+
+    # Move files to sent directory
+    logger.info('Moving to sent files directory', file_paths=file_paths)
+    for sent_file in file_paths:
+        sent_file.replace(Config.SENT_FILES_DIRECTORY.joinpath(sent_file.name))
 
 
-def check_files(partial_files_dir: Path, sent_files_dir: Path):
+def check_files(partial_files_dir: Path):
     for print_file in partial_files_dir.rglob('*'):
         file_name_parts = print_file.name.split('.')
         expected_number_of_lines = int(file_name_parts[3])
+        pack_code = file_name_parts[1]
         actual_number_of_lines = sum(1 for _ in print_file.open())
-
         if expected_number_of_lines == actual_number_of_lines:
-            process_complete_file(print_file)
-            print_file.replace(sent_files_dir.joinpath(print_file.name))
-            logger.info(f"File hero has encrypted the file [{print_file}],"
-                        f" created a manifest and SFTP'ed the two files")
+            process_complete_file(print_file, pack_code)
+            logger.info('Removing unencrypted file', file_name=print_file.name)
+            os.remove(print_file.as_posix())
 
 
 def start_file_sender(readiness_queue):
     # TODO Connect to SFTP etc. before readying
     readiness_queue.put(True)
     logger.info('Started file sender')
     while True:
-        check_files(Config.PARTIAL_FILES_DIRECTORY, Config.SENT_FILES_DIRECTORY)
+        check_files(Config.PARTIAL_FILES_DIRECTORY)
         sleep(2)
+
+
+def copy_files_to_sftp(file_paths: Collection[Path]):
+    with sftp.SftpUtility() as sftp_client:
```

Comment:
The SFTP directory needs to be different for different suppliers, needs to be mapped from the pack code